### PR TITLE
Orca: add python 3.8 nightly test for xshards

### PR DESCRIPTION
**Pull request by hamham223**
### 1. Why the change?

Next step of https://github.com/intel-analytics/BigDL/pull/7870

Now all the python 3.8 tests are tested in `Nightly test` !

### 2. Summary of the change

+ modify the tutorial notebooks to tensorflow 2 uts
+ triggers (`nightly_test.yml`, `action.yml`), use tensorflow==2.9.0

### 3. How to test?

The following nightly-test

- [x] Orca-Tutorial-Notebook-Xshards-Py38-Spark3
- [x] Orca-Tutorial-Notebook-Xshards-Image-Py38-Spark3
- [x] Orca-Tutorial-Notebook-Xshards-Py37-Spark3

PR test:
- [x] Orca-Tutorial-Notebook-Xshards-Py37-Spark3

### CODE REVIEW

File: .github/actions/orca/orca-tutorial-notebook-xshards-image-py38-spark3-action/pr-test/action.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  using: "composite"
+  steps:
+    - name: Run Test
+      shell: bash
+      run: |
+        source activate py38
+        pip uninstall bigdl-orca
+        pip uninstall bigdl-dllib
+        export SPARK_LOCAL_HOSTNAME=localhost
+        bash python/dev/release_default_linux_spark2.sh default false false false -Ddata-store-url=$HTTP_URI -U
```
[the remark this answers is not in the document]
Comment (hkvision) [reply .github/actions/orca/orca-tutorial-notebook-xshards-image-py38-spark3-action/pr-test/action.yml:13]: yes, should use spark3; also need to check if other tests are using spark2.

File: python/orca/tutorial/xshards/notebooks/diabetes.ipynb [hunk 2]
```diff
@@ -352,7 +354,8 @@
    "source": [
     "est.fit(data=data_shards,\n",
     "        batch_size=16,\n",
-    "        epochs=150)"
+    "        epochs=150,\n",
+    "        steps_per_epoch=1)"
```
[the remark this answers is not in the document]
Comment (hkvision) [reply python/orca/tutorial/xshards/notebooks/diabetes.ipynb:358]: I think you need to add the calculation of steps_per_epoch to the script.

File: python/orca/tutorial/xshards/notebooks/diabetes.ipynb [hunk 3]
```diff
@@ -376,9 +382,9 @@
  ],
  "metadata": {
   "kernelspec": {
-   "display_name": "py37tf2_x",
+   "display_name": "yinchen",
```
Comment (hkvision) [inline python/orca/tutorial/xshards/notebooks/diabetes.ipynb:385]: revert this?

Comment (hamham223) [reply python/orca/tutorial/xshards/notebooks/diabetes.ipynb:385]: oh no, vscode automatically change this 😢

File: python/orca/tutorial/xshards/notebooks/tabular_playground_series.ipynb [hunk 4]
```diff
@@ -506,7 +506,7 @@
...
...af86S5I0QYaIJKmbIbLIOI3LwkmyPsnWJDdNupdpl+SQJFck2ZxkU5K3T7qnaZZk3yRXJ/le+zzfO+meenlOZBFp07j8N/ASYAuDq89OrqrNE21sSiV5EfAAcF5VHTbpfqZZkoOBg6vqu0meBFwLnOh/m32SBPi9qnogyd7AN4G3V9VVE27tUfNIZHFxGpcFVFVXAndPuo/dQVXdWVXfbcs/A24Glk22q+lVAw+01b3bYyr/j94QWVyWAbcPrW/Bf6haZJKsAJ4HfGfCrUy1JEuSXA9sBTZU1VR+noaIpHlL8kTgQuAdVXX/pPuZZlX1UFUdzmCmjSOTTOVXrobI4uI0Llq02nf3FwKfq6ovT7qf3UVV3QtcAayZcCtdDJHFxWlctCi1E8HnADdX1Ucm3c+0SzKTZP+2/HgGF9P810Sb6mSILCJVtR3YMY3LzcAFu8k0LhOR5PPAt4FnJdmS5NRJ9zTFjgbeCByT5Pr2OH7STU2xg4ErktzA4H8eN1TVJRPuqYuX+EqSunkkIknqZohIkroZIpKkboaIJKmbISJJ6maISAsoyf5J3jKG1zkxyapRv470uxgi0sLaH5h3iGSg59/hiYAhoonzdyLSAkqyY+blWxhMZfEc4AAGs7T+Q1Vd1CYwvIzBBIZHAMcDpwBvALYxmITz2qr6cJJnAGcBM8AvgDcDTwEuAe5rj1dX1ffH9R6lYUsn3YC0mzkdOKyqDk+yFHhCVd2f5EDgqiQ7prFZCaytqquS/DHwauC5DMLmuwzu1wFwNvA3VXVrkhcAn6yqY9rzXFJVXxrnm5N2ZohIoxPgn9vNsX7NYFr/g9q2Hw3dgOho4KKqehB4MMlX4OEZc18IfHEwdRUAjxtX89J8GCLS6LyewddQR1TV/yb5IbBv2/bzeey/F3Bvmy5cWpQ8sS4trJ8BT2rLTwa2tgD5M+Dpu9jnW8Ar2n23nwi8HKDdr+MHSV4LD5+Ef+4cryNNjCEiLaCqugv4VpKbgMOB1UluZHDifM6pvqvqGgZT/t8AfBW4kcEJcxgczZya5HvAJn5zu+TzgXcmua6dfJcmwquzpEUgyROr6oEkTwCuBNbtuKe5tJh5TkRaHM5uPx7cFzjXANG08EhEktTNcyKSpG6GiCSpmyEiSepmiEiSuhkikqRu/wc5Xa805Nl9GwAAAABJRU5ErkJggg==\n",
```
Comment (hkvision) [inline python/orca/tutorial/xshards/notebooks/tabular_playground_series.ipynb:509]: Does this change affects anything?

**Comment by hkvision** [comment]
Also need to change the py files into tf2. Maybe in a next PR.

**Review by hkvision — APPROVED** [review]

File: python/orca/tutorial/xshards/notebooks/tabular_playground_series.ipynb (hunk 4 above)
Comment (hamham223) [reply python/orca/tutorial/xshards/notebooks/tabular_playground_series.ipynb:509]: seemly not.. let me try to eliminate it

File: python/orca/tutorial/xshards/notebooks/auto_mpg.ipynb [hunk 5]
```diff
@@ -797,24 +798,14 @@
...
    ]
   },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "ac0db7e4",
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "model = build_model()"
```
Comment (hamham223) [inline python/orca/tutorial/xshards/notebooks/auto_mpg.ipynb:807]: Using `build_model()` as the model creator function.
This code block is not used, thus removed. This is due to hamster not due to vscode 😢

**Pull request merged**